### PR TITLE
Implement [M3C] Sage of the Maze

### PR DESCRIPTION
**Pull request by grimreap124**
https://scryfall.com/card/m3c/67/sage-of-the-maze

### CODE REVIEW

File: Mage.Sets/src/mage/cards/s/SageOfTheMaze.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+}
+
+class SageOfTheMazeEffect extends OneShotEffect {
+
+    private static final FilterPermanent filter
+            = new FilterControlledPermanent(SubType.GATE, "Gates you control");
+
+    private static final DynamicValue xValue = new PermanentsOnBattlefieldCount(filter, 2);
+    private static final Hint hint = new ValueHint("Gates you control", xValue);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/s/SageOfTheMaze.java:79]: looks like you didn't add the hint anywhere (need to provide a getter or just drop the private modifier)

File: Mage.Sets/src/mage/cards/s/SageOfTheMaze.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // {T}: Add two mana in any combination of colors.
+        Ability ability = new SimpleManaAbility(Zone.BATTLEFIELD, new AddManaInAnyCombinationEffect(2), new TapSourceCost());
+        this.addAbility(ability);
+
+        // {T}: Until end of turn, target land you control becomes an X/X Citizen creature with haste in addition to its other types, where X is twice the number of Gates you control. Activate only as a sorcery.
+        ability = new SimpleActivatedAbility(new SageOfTheMazeEffect(), new TapSourceCost());
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/s/SageOfTheMaze.java:52]: this one should be as a sorcery, not the other

**Review by xenohedron — COMMENTED** [review]
verify failure is unrelated

File: Mage.Sets/src/mage/cards/s/SageOfTheMaze.java (hunk 1 above)
Comment (grimreap124) [reply Mage.Sets/src/mage/cards/s/SageOfTheMaze.java]: dropped the private modifier and added it to the ability

File: Mage.Sets/src/mage/cards/s/SageOfTheMaze.java (hunk 2 above)
Comment (grimreap124) [reply Mage.Sets/src/mage/cards/s/SageOfTheMaze.java]: Ooops I must have updated the wrong ability to be a sorcery. Updated

**Review by xenohedron — APPROVED** [review]

**Pull request merged**